pkg/common/providers/rosaprovider/cluster.go

```diff
@@ -121,6 +122,19 @@ func (m *ROSAProvider) LaunchCluster(clusterName string) (string, error) {
 		"--pod-cidr", viper.GetString(PodCIDR),
 		"--host-prefix", viper.GetString(HostPrefix),
 	}
+	if viper.GetString(SubnetIDs) != "" {
+		subnetIDs := viper.GetString(SubnetIDs)
+		createClusterArgs = append(createClusterArgs, "--subnet-ids", subnetIDs)
+		if httpProxy := viper.GetString(config.Proxy.HttpProxy); httpProxy != "" {
+			createClusterArgs = append(createClusterArgs, "--http-proxy", httpProxy)
+		}
+		if httpsProxy := viper.GetString(config.Proxy.HttpsProxy); httpsProxy != "" {
+			createClusterArgs = append(createClusterArgs, "--https-proxy", httpsProxy)
+		}
+		if userCaBundle := viper.GetString(config.Proxy.UserCABundle); userCaBundle != "" {
+			createClusterArgs = append(createClusterArgs, "--additional-trust-bundle-file", userCaBundle)
```

Comment:
There's a slight issue here we need to figure out.

The value of `userCABundle` here is derived from the `USER_CA_BUNDLE` config environment variable.

Currently, we have two ways of handling that value:
- it either represents a filepath preceded by an @, in which case the contents are loaded into memory.
- or it is the actual text representation of the CA bundle.

The problem that's happening is that this code demands that `USER_CA_BUNDLE` be a filepath, but not one prefixed by an `@` to load the contents into memory.

Which is fine, except that it then breaks the other dat-2 proxy tests we just introduced, as they would attempt to set a user CA bundle on clusters which is a value that's just a filepath, rather than the contents of the file.

So we need to align our strategy here so that one doesn't break the other.

- Only ever treat `USER_CA_BUNDLE` as a filepath (no @ prefix). In which case the day-2 tests need to be fixed to load the contents of this path rather than just take the value in `USER_CA_BUNDLE`.
- Have `USER_CA_BUNDLE` be either a filepath (with @ prefix) or the actual CA value (like it currently behaves). In which case this code needs to be fixed to write the contents of `config.Proxy.UserCABundle` to a temporary file which is passed to the calling arguments here.
- Something else..?

Reply:
Ok, so the current problem:
If we set the USER_CA_BUNLDE=@<path_ca_file>
It works in ocmprovider during installation, since the `AdditionalTrustBundle()` func in the `ClusterBuilder` accepts the value of the file only.
But it does not in rosaprovider during installation, since it appends the value directly to the cmd. `rosa create cluster --additional-trust-bundle-file`

If we set the USER_CA_BUNDLE=<path_ca_file> without `@`
It works in rosaprovider during installation, but after that, the `AddClusterProxy()`, `RemoveClusterProxy()` and `RemoveUserCABundle()` in package ocmprovider will be failed, since the USER_CA_BUNDLE will be treated as contents. And rosaprovider is referring to the same functions in ocmproviders above.

If we set the USER_CA_BUNDLE=<ca_contents>
If will not work for rosaprovider as we do not have a way to inject the contents of the ca certs to `rosa create` command now.

My thoughts:
1. We move the `LoadUserCABundle()` func out side the config.go, so we can process the `USER_CA_BUNDLE=@<file>` for ocmprovider during installation time only. (Seems the same approach with your 1st point?)
2. Add a new environment like `USER_CA_BUNDLE_FILE` which used for rosa installation. We can also generate the contents value of user ca bundle based on this, but seems increasing the complexity.
3. Write contents to temporary file is not a good idea IMO...

Personally, I think we always treat the value of `USER_CA_BUNDLE` as a filepath, and abstract the contents when needed is better approach. Unless we have a use case that we need to pass the environment as the contents value of the CA.

Reply:
Thanks for doing this change, I tested it and it is working well.